packages/repository/src/__tests__/integration/repositories/relation.factory.integration.ts

```diff
@@ -203,6 +211,245 @@ describe('BelongsTo relation', () => {
   }
 });
 
+describe('HasManyThrough relation', () => {
+  let existingCustomerId: number;
+  // Customer has many CartItems through CustomerCartItemLink
+  let hasManyThroughRepo: HasManyThroughRepository<
+    CartItem,
+    typeof CartItem.prototype.id,
+    CustomerCartItemLink
+  >;
+  let hasManyThroughFactory: HasManyThroughRepositoryFactory<
+    CartItem,
+    typeof CartItem.prototype.id,
+    CustomerCartItemLink,
+    typeof Customer.prototype.id
+  >;
+
+  before(givenCrudRepositories);
+  before(givenPersistedCustomerInstance);
+  before(givenConstrainedRepositories);
+
+  beforeEach(async function resetDatabase() {
+    await customerRepo.deleteAll();
+    await CustomerCartItemLinkRepo.deleteAll();
+    await cartItemRepo.deleteAll();
+  });
+
+  it('creates an target instance alone with the corresponding through model', async () => {
+    const cartItem = await hasManyThroughRepo.create(
+      {
+        description: 'an item hasManyThrough',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const persistedItem = await cartItemRepo.findById(cartItem.id);
+    const persistedLink = await CustomerCartItemLinkRepo.find();
+    expect(cartItem).to.deepEqual(persistedItem);
+    expect(persistedLink).have.length(1);
+    const expected = {
+      id: 99,
+      customerId: existingCustomerId,
+      itemId: cartItem.id,
+    };
+    expect(toJSON(persistedLink[0])).to.deepEqual(toJSON(expected));
+  });
+
+  it('finds an instance via through model', async () => {
+    const item = await hasManyThroughRepo.create(
+      {
+        description: 'an item hasManyThrough',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const notMyItem = await cartItemRepo.create({
+      description: "someone else's item desc",
+    });
+
+    const items = await hasManyThroughRepo.find();
+
+    expect(items).to.not.containEql(notMyItem);
+    expect(items).to.deepEqual([item]);
+  });
+
+  it('finds instances via through models', async () => {
+    const item1 = await hasManyThroughRepo.create(
+      {
+        description: 'group 1',
+      },
+      {
+        throughData: {id: 99},
+      },
+    );
+    const item2 = await hasManyThroughRepo.create(
+      {
+        description: 'group 2',
+      },
+      {
+        throughData: {id: 98},
+      },
+    );
+    const items = await hasManyThroughRepo.find();
+
+    expect(items).have.length(2);
+    expect(items).to.deepEqual([item1, item2]);
+    const group1 = await hasManyThroughRepo.find({
+      where: {description: 'group 1'},
+    });
+    expect(group1).to.deepEqual([item1]);
+  });
+
+  it('deletes an instance then delete the through model', async () => {
```

Comment:
`deletes an instance then delete the through model` -> `deletes an instance, then deletes the through model`.